src/View/Helper/PaginatorHelper.php

```diff
@@ -463,9 +480,14 @@ public function generateUrl(array $options = [], $model = null, $full = false)
             'sort' => $paging['sort'],
             'direction' => $paging['direction'],
         ];
-
-        if (!empty($this->_config['options']['url'])) {
-            $url = array_merge($url, $this->_config['options']['url']);
+        if (empty($paging['prefix'])) {
+            if (!empty($this->_config['options']['url'])) {
+                $url = array_merge($url, $this->_config['options']['url']);
+            }
+        } else {
+            if (!empty($this->_config['options']['url'][$paging['prefix']])) {
```

Comment:
If I did this *and* the above change, I think the paging prefix will never be checked if there is another paginator invoked without a prefix.